### PR TITLE
Implement SectionResolver

### LINKED ISSUE #80: Implement SectionResolver to get sections from articles
SectionResolver gets sections in articles by heading. It is used for the `wikibase-section` class (called `wikipedia-paragraphs` in Ruby WikipediaBase).

A skeleton implementation was started in wikipediabase/resolvers/section.py.

### PR DESCRIPTION
**Pull request by chsamlee**
Resolves #80 (no pun intended)

### CODE REVIEW

**Comment by sherryy** [comment]
Looks good to me (but wait for the travis style check to complete)

File: wikipediabase/resolvers/section.py [hunk 1]
```diff
@@ -1,5 +1,13 @@
+import re
+
+import bs4
+
```
Comment (alvaromorales) [inline wikipediabase/resolvers/section.py]: remove empty line 4

the convention for imports is:

standard library imports
    < newline >
third-party library imports
    < newline >
this project's imports

File: wikipediabase/resolvers/section.py [hunk 2]
```diff
@@ -12,12 +20,101 @@ class SectionResolver(BaseResolver):
...
+        elif isinstance(raw, bs4.element.Tag):
+            # remove edit button
+            edit_btn = raw.find(class_="mw-editsection")
+            if edit_btn:
+                edit_btn.extract()
+            # remove references
+            for ref in raw.find_all("sup", class_="reference"):
+                ref.extract()
+            # make URLs absolute
+            base_url = "http://en.wikipedia.org"
```
Comment (alvaromorales) [inline wikipediabase/resolvers/section.py]: use `https://`

File: wikipediabase/resolvers/section.py [hunk 3]
```diff
@@ -12,12 +20,101 @@ class SectionResolver(BaseResolver):
...
+                    link["href"] = "%s%s" % (base_url, url)
+                elif url.startswith("#"):
+                    link["href"] = "%s%s" % (article_url, url)
+            # convert to unicode and remove line breaks between tags
+            txt = unicode(raw)
+            txt = txt.replace(">\n<", "><")
+            return txt
+
+        else:
+            raise TypeError("Unknown bs4 type!")
```
Comment (alvaromorales) [inline wikipediabase/resolvers/section.py]: this exception is not being caught in `resolve_section`

**Comment by alvaromorales** [comment]
Very impressive! Have you done enough testing with the Random Article button?

**Comment by alvaromorales** [comment]
lgtm :rocket:

**Pull request merged**